src/coreclr/jit/llvm.cpp

```diff
@@ -1752,6 +1766,10 @@ void Llvm::populateLlvmArgNums()
         {
             varDsc->lvLlvmArgNum  = nextLlvmArgNum++;
             varDsc->lvCorInfoType = corInfoType;
+            if (classHnd != NO_CLASS_HANDLE && !varTypeIsStruct(varDsc))
+            {
+                varDsc->lvClassHnd = classHnd;
+            }
```

Comment:
We could use `lvClassHnd` to store the exact class always (not just for non structs). Seems that would be a simpler model?

Reply:
yes, that's better thanks.